chapters/unitexpressions.tex

```diff
@@ -89,3 +89,133 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
+%  \end{itemize}
+\end{itemize}
+
+
+\subsection{Expressions with Empty Unit}\label{expressions-with-empty-unit}
+
+This section describes conditions under which an expression has empty unit.
+Conditions not given here must not be interpreted as definitely not implying empty unit; instead, the unit may be currently undefined for some expressions, allowing the unit to be properly defined in future versions of the specification.
+
+Basic expressions defined to have empty unit:
+\begin{itemize}
+\item
+  \lstinline!Real! literals.
+\item
+  \lstinline!Integer! expressions implicitly cast to \lstinline!Real!.
+\end{itemize}
+
+Expressions defined to \emph{not} propagate the empty unit, thereby forcing inference of unit \lstinline!"1"!:
+\begin{itemize}
+\item
+  Addition, subtraction, multiplication and division operators when either operand has non-empty unit.
+\item
+  Right operand of binary exponentiation.
+\item
+  Component references outside functions where the declared \lstinline!unit!-attribute is \lstinline!""! (possibly by not being specified).
+  (Unit checking involving user-defined functions with empty unit on inputs and outputs is currently not defined.)
+\end{itemize}
+
+Built-in non-array operators, functions, and special expressions that propagate any unit (including empty):
+\begin{itemize}
+\item
+  Negation, addition, subtraction (scalar or element-wise)
+\end{itemize}
+
+Special situations in which the empty unit can be propagated up the expression tree:
+\begin{itemize}
+\item
+  Multiplication and division when both operands have empty unit.
+\end{itemize}
+
+\begin{example}
+Consider unit checking of the following declaration equation:
+\begin{lstlisting}[language=modelica]
+Real y(unit = "m") = 1 + 2.5 * 3;
+\end{lstlisting}
+The unit of the declaration equation right-hand side is determined as follows:
+\begin{itemize}
+\item The \lstinline!Real! literal \lstinline!2.5! has empty unit.
+\item The \lstinline!Integer! literals \lstinline!1! and \lstinline!3! are implicitly cast to \lstinline!Real!, and therefore have empty unit.
+\item The multiplication \lstinline!2.5 * 3! has empty unit, as multiplication can propagate the empty unit of both operands.
+\item The addition \lstinline!1 + (2.5 * 3)! has empty unit, as addition can propagate any unit as long as both operands have the same unit.
+\item The entire right-hand side expression gets inferred unit \lstinline!"m"! in order to be compatible with the component's declared \lstinline!unit!.
+\end{itemize}
+\end{example}
+
+\begin{example}
+Consider unit checking of the following erroneous declaration equation for \lstinline!y!:
+\begin{lstlisting}[language=modelica]
+Real x(unit = "m") = 1.0;
+Real y(unit = "m") = x^2 / 2;
+\end{lstlisting}
+The unit of the declaration equation right-hand side is determined as follows:
+\begin{itemize}
+\item The unit of \lstinline!x! is \lstinline!"m"!, and common sense gives that \lstinline!x^2! has unit \lstinline!"m2"!
```

Comment:
No common sense needed, dimensional analysis is enough. 😁

Reply:
Yes, calling it _standard dimensional analysis_ sounds better without actually changing the meaning.

Perhaps we could find a standard reference on dimensional analysis that can be mentioned in the introduction to the section/chapter.  Maybe one among the references listed on https://en.wikipedia.org/wiki/Dimensional_analysis?

Reply:
I think there will be better places to attach a discussion about which reference to give, so maybe we can resolve this one to start with?

Reply:
I'd just call it "dimensional analysis"; it's not standardized by some document afaik, and I wouldn't know what "non-standard dimensional analysis" would be. :-)

Reply:
OK, I'll resolve this, but I am totally opening to continuing the discussion about "standard" elsewhere.

The reason why I'd like to qualify the term with "standard" is that I want to indicate that this is not a concept fully developed within the Modelica Specification, but that it is easy to find external resources that explain the concept.  Let's think about other ways of expressing this…

Reply:
> I'd just call it "dimensional analysis"; it's not standardized by some document afaik, and I wouldn't know what "non-standard dimensional analysis" would be. :-) 

Dimensional analysis for Non-standard Analysis (as re-introduced by A. Robinson)? :-)

Reply:
> > I'd just call it "dimensional analysis"; it's not standardized by some document afaik, and I wouldn't know what "non-standard dimensional analysis" would be. :-)
> 
> Dimensional analysis for Non-standard Analysis (as re-introduced by A. Robinson)? :-)

Ha, learned something new today. 😁